### ISSUE TITLE
[Ray Java] The actor died unexpectedly before finishing this task 

### ISSUE THREAD

**Issue by ysu17** [ray-project/ray#34993]
### What happened + What you expected to happen

I am experimenting with the Java bindings for Ray. In my Java unit test, I am using the example shown on the [Ray Core Quick Start](https://docs.ray.io/en/latest/ray-overview/getting-started.html#ray-core-quick-start), however, I always get the "The actor died unexpectedly before finishing this task" error:
```
io.ray.api.exception.RayActorException: The actor f5a58f005c97426e995b79d601000000 died unexpectedly before finishing this task.
        at io.ray.runtime.object.ObjectSerializer.deserializeActorException(ObjectSerializer.java:257)
        at io.ray.runtime.object.ObjectSerializer.deserialize(ObjectSerializer.java:104)
        at io.ray.runtime.object.ObjectStore.get(ObjectStore.java:140)
        at io.ray.runtime.AbstractRayRuntime.get(AbstractRayRuntime.java:144)
        at io.ray.runtime.AbstractRayRuntime.get(AbstractRayRuntime.java:131)
        at io.ray.api.Ray.get(Ray.java:120)
```
I checked the `gcs_server.out` log, and found that the actor died because the worker exited:
```
[2023-05-02 18:13:31,914 W 6113 6113] (gcs_server) gcs_worker_manager.cc:44: Reporting worker exit, worker id = f32818541cdb0fe0d584a98df1d5b857b1d78e77e6771051ba168356, node id = fa83b4a4a82e9cbd688aee18ad9134c7174f64837952a7d812015888, address = 10.32.131.231, exit_type = SYSTEM_ERROR, exit_detail = Worker unexpectedly exits with a connection error code 2. End of file. There are some potential root causes. (1) The process is killed by SIGKILL by OOM killer due to high memory usage. (2) ray stop --force is called. (3) The worker is crashed unexpectedly due to SIGSEGV or other unexpected errors.. Unintentional worker failures have been reported. If there are lots of this logs, that might indicate there are unexpected failures in the cluster.
[2023-05-02 18:13:31,914 W 6113 6113] (gcs_server) gcs_actor_manager.cc:912: Worker f32818541cdb0fe0d584a98df1d5b857b1d78e77e6771051ba168356 on node fa83b4a4a82e9cbd688aee18ad9134c7174f64837952a7d812015888 exits, type=SYSTEM_ERROR, has creation_task_exception = 0
[2023-05-02 18:13:31,914 I 6113 6113] (gcs_server) gcs_actor_manager.cc:1083: Actor ad1299f17819b83905df732901000000 is failed on worker f32818541cdb0fe0d584a98df1d5b857b1d78e77e6771051ba168356 at node fa83b4a4a82e9cbd688aee18ad9134c7174f64837952a7d812015888, need_reschedule = 1, death context type = ActorDiedErrorContext, remaining_restarts = 0, job id = 01000000
[2023-05-02 18:13:32,003 I 6113 6113] (gcs_server) gcs_actor_manager.cc:755: Destroying actor, actor id = ad1299f17819b83905df732901000000, job id = 01000000
```
I did not see any OOM error and I did not run `ray stop --force`, so it seems the worker crashed due to some unexpected errors. 

In the `java-core-driver` log, I see the following:
```
[2023-05-02 18:13:31,914 I 6060 6320] direct_actor_task_submitter.cc:563: PushActorTask failed because of network error, this task will be stashed away and waiting for Death info from GCS, task_id=c2668a65bda616c1ad1299f17819b83905df732901000000, wait_queue_size=1
```
I could not see a clue about exactly what caused the worker crash. Not sure if this is related to the configuration, but seems the Ray configuration support for [Java applications](https://docs.ray.io/en/latest/ray-core/configure.html#java-applications) is limited, and I am not sure how to tune it with Ray Java. Any suggestions on how I can resolve this? Thanks.

### Versions / Dependencies

OS: CentOS 7
Java: 11.0.8-msft
Python: 3.7
Ray: 2.2
ray-api: 2.4.0 (re-built on our own host)
ray-runtime: 2.4.0 (re-built on our own host)

### Reproduction script

The code is the same to the example shown on the [Ray Core Quick Start](https://docs.ray.io/en/latest/ray-overview/getting-started.html#ray-core-quick-start), i.e. 
```
public void testRayActor() {
    // Intialize Ray runtime.
    Ray.init();

    List<ActorHandle<Counter>> counters = new ArrayList<>();
    // Create 4 actors from the `Counter` class.
    // They will run in remote worker processes.
    for (int i = 0; i < 4; i++) {
      counters.add(Ray.actor(Counter::new).remote());
    }
    // Invoke the `increment` method on each actor.
    // This will send an actor task to each remote actor.
    for (ActorHandle<Counter> counter : counters) {
      counter.task(Counter::increment).remote();
    }
    // Invoke the `read` method on each actor, and print the results.
    List<ObjectRef<Integer>> objectRefList = counters.stream()
        .map(counter -> counter.task(Counter::read).remote())
        .collect(Collectors.toList());
    System.out.println(Ray.get(objectRefList));  // [1, 1, 1, 1]
  }

  public static class Counter {
    private int value = 0;

    public void increment() {
      this.value += 1;
    }

    public int read() {
      return this.value;
    }
  }
```

### Issue Severity

High: It blocks me from completing my task.

**Comment by SongGuyang** [comment]
Any other error logs from *.err log files?

**Comment by XiaodongLv** [comment]
I will check it.

**Comment by jovany-wang** [comment]
Hi @ysu-linkedin , it seems there's some thing wrong in java actor. Please upload the related log files: `java-core-worker, `java-worker`.

**Comment by XiaodongLv** [comment]
I‘ve tried locally，but it isn’t reproduced.
As the following steps：
1、pip install ray==2.2.0
2、RayDemo.java
package com.lxd;
import io.ray.api.ActorHandle;
import io.ray.api.ObjectRef;
import io.ray.api.Ray;
import java.util.ArrayList;
import java.util.List;
import java.util.stream.Collectors;

public class RayDemo {

    public static class Counter {

        private int value = 0;

        public void increment() {
            this.value += 1;
        }

        public int read() {
            return this.value;
        }
    }

    public static void main(String[] args) {
        Ray.init();
        List<ActorHandle<Counter>> counters = new ArrayList<>();

        for (int i = 0; i < 4; i++) {
            counters.add(Ray.actor(Counter::new).remote());
        }

        for (ActorHandle<Counter> counter : counters) {
            counter.task(Counter::increment).remote();
        }

        List<ObjectRef<Integer>> objectRefList = counters.stream()
            .map(counter -> counter.task(Counter::read).remote())
            .collect(Collectors.toList());
        System.out.println(Ray.get(objectRefList));  // [1, 1, 1, 1]
    }
}
3、pom.xml
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
  <modelVersion>4.0.0</modelVersion>
  <groupId>com.lxd</groupId>
  <artifactId>RayDemo</artifactId>
  <packaging>jar</packaging>
  <version>1.0.0</version>
  <properties>
    <ray.version>2.2.0</ray.version>
    <ray.group>io.ray</ray.group>
  </properties>
  <build>
    <plugins>
      <plugin>
        <groupId>org.apache.maven.plugins</groupId>
        <artifactId>maven-compiler-plugin</artifactId>
        <configuration>
          <source>8</source>
          <target>8</target>
        </configuration>
      </plugin>
      <plugin>
        <groupId>org.apache.maven.plugins</groupId>
        <artifactId>maven-assembly-plugin</artifactId>
        <version>3.0.0</version>
        <configuration>
          <archive>
            <manifest>
              <mainClass>com.lxd.RayDemo</mainClass>
            </manifest>
          </archive>
          <descriptorRefs>
            <descriptorRef>jar-with-dependencies</descriptorRef>
          </descriptorRefs>
        </configuration>
        <executions>
          <execution>
            <id>make-assembly</id> 
            <phase>package</phase> 
            <goals>
              <goal>single</goal>
            </goals>
          </execution>
        </executions>
      </plugin>
    </plugins>
  </build>
  <name>RayDemo</name>
  <url>http://maven.apache.org</url>
  <dependencies>
    <dependency>
      <groupId>junit</groupId>
      <artifactId>junit</artifactId>
      <version>3.8.1</version>
      <scope>test</scope>
    </dependency>
    <dependency>
      <groupId>io.ray</groupId>
      <artifactId>ray-api</artifactId>
      <version>${ray.version}</version>
</dependency>
    <dependency>
      <groupId>io.ray</groupId>
      <artifactId>ray-runtime</artifactId>
      <version>${ray.version}</version>
    </dependency>
  </dependencies>
</project>

It can run normally, maybe you can upload your project or zip of dir which is /tmp/ray/session_latest/logs

**Comment by ysu17** [comment]
Update: this issue is resolved by upgrading ray from 2.2 to 2.4, which is built from source on our own machine.
